app/controllers/admin/pet_applications_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::PetApplicationsController < ApplicationController
+  def update
+    pet_application = PetApplication.find(params[:id])
+
+    pet_application.update(status: "#{params[:status]}")
+
+    redirect_to "/admin/applications/#{params[:application_id]}"
+  end
+end
```

Comment:
I talked about this in a couple of places already but once your route is set up properly, the code in this method action would look similar, only line 3 would need to change.

```ruby
pet_application = PetApplication.find_by(application_id: params[:application_id], pet_id: params[:id])
```